lib/exceptions.rb

```diff
@@ -170,6 +170,7 @@ class UpgradeFailed < MachineryError; end
     class BinaryDiffError < MachineryError; end
 
     class SshConnectionFailed < MachineryError; end
+    class SshConnectionDisrupted < SystemExit; end
```

Comment:
But we also have the problem of dependencies in every other case where on inspector failed, so I think we could just handle this case the same.